pineappl_cli/src/export/applgrid.rs

```diff
@@ -100,11 +124,15 @@ pub fn convert_into_applgrid(
     ffi::make_lumi_pdf(id, &combinations).into_raw();
 
     let limits = &bin_info.limits();
-    let limits: Vec<_> = limits
-        .iter()
-        .map(|vec| vec[0].0)
-        .chain(limits.last().map(|vec| vec[0].1))
-        .collect();
+    let limits = if integer_bin_limits {
+        (0..=limits.len()).map(|x| x as f64).collect::<Vec<_>>()
```

Comment:
Here there's very likely a normalization factor missing. If you change the bin sizes you must correct the cross sections.

Reply:
You're right, thanks. Does APPLgrid also divide by the bin widths in the end?